play/src/main/scala/com/gu/pandahmac/HmacSecrets.scala

```diff
@@ -0,0 +1,28 @@
+package com.gu.pandahmac
+
+import com.gu.hmac.ValidateHMACHeader
+
+import java.net.URI
+
+trait HMACSecrets extends ValidateHMACHeader {
+  @deprecated("In an upcoming release consumers will expected to provide a list of secrets to `secretKeys` to allow for secret rotation.")
+  def secret: String = ""
+  def secretKeys: List[String] = List.empty
+
+  protected def gatherSecrets = {
+    val gatheredSecrets = (secret, secretKeys) match {
+      case ("", emptyList) if emptyList.isEmpty => List.empty
+      case (nonEmptySecret, _) if !nonEmptySecret.isEmpty => List(nonEmptySecret)
+      case ("", nonEmptyList) => nonEmptyList
```

Comment:
I think these cases should be swapped? (if both are set, we shouldn't prefer the deprecated one)